api/services/DMStore.ts

```diff
@@ -287,13 +298,65 @@ export async function getInfo(): Promise<JSON> {
     return response.data
 }
 
+function getOptions(req) {
+    return headerUtilities.getAuthHeadersWithUserIdAndRoles(req)
+}
+
+/**
+ * uploadDocumentAndAssociateWithCase
+ *
+ * Upload the document to Document Management Store.
+ *
+ * Once the document has been uploaded, get the Event Token to update a Case, create the payload with that
+ * Event Token, and update the Case with a link to the document.
+ *
+ * TODO: Once a service line has implemented upload into their Case Definitions file, use prepareCaseForUpload
+ * within ccdDataStoreApiPayloads.ts and check that it works correctly with a service line.
+ */
+async function uploadDocumentAndAssociateWithCase(userId, caseId, jurisdiction, eventId, caseType, file, fileNotes,
```

Comment:
can we use axios and remove getoptions as roles are being sent as default, instead of passing options maybe pass userID and use it as  custom header option

Reply:
Unfortunately we couldn't use Axios as yet, as I couldn't get it working with posting files within formdata, therefore I'm passing in an options object which contains header data, ie. auth + service auth to make the promise-request call. But all promise-request logic is in a separate file, so it should be easy to remove.